### ISSUE TITLE
ConcurrencyException with Sitemaps when saving two content items at once

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#15743]
### Describe the bug

With the Sitemaps module enabled, you can get `ConcurrencyException` when saving two content items that are in the sitemaps simultaneously.

### To Reproduce
1. Enable and configure the Sitemaps module.
2. Publish two content items at the same time.
3. Experience "ConcurrencyException: The document could not be updated as it has been changed by another process"

### Expected behavior
Both publishes just work.

This is because [`ContentTypesSitemapUpdateHandler` runs for every publish](https://github.com/OrchardCMS/OrchardCore/blob/b1b47175d438974813d040fbcca4f2e6764d47f2/src/OrchardCore.Modules/OrchardCore.Contents/Sitemaps/ContentTypesSitemapUpdateHandler.cs#L16) and updates the same single document. I guess the same issue can happen with simultaneous content item removals due to the same reason.

This can be fixed by putting sitemap updates in `SitemapManager` within a distributed lock, and instead of starting updates from a content handler, just adding a deferred task and doing the update there. So, updates would be synchronized without an adverse effect on the UX.

### Screenshots

![image (1) (4)](https://github.com/OrchardCMS/OrchardCore/assets/1976647/893d4c15-beb3-4229-8dea-ee735940dcf5)

**Comment by MikeAlhayek** [comment]
@Piedone can you please submit a PR for this?
 
Too many 1.9 related issues that should be soon addressed.

**Comment by Piedone** [comment]
Yeah, I intend to work on this soon. Is this also a post-1.8 regression though?

**Comment by MikeAlhayek** [comment]
1.8 has been out for a while and I don't hear people screaming over it. We could back port it into 1.8 release branch and release 1.8.3 if needed.

I think once you submit a PR. We could release 1.8.3 if we end up not release 1.9 soonish

**Comment by Piedone** [comment]
Ah OK, just because you were asking about a PR in context of 1.9 issues.

But now it comes to me that this is definitely not a post-1.8 regression, since I have this with a client project on 1.7 (but the affected code is the same in `main` too).
